DvdLib/Ifo/Dvd.cs

```diff
@@ -19,17 +19,20 @@ public Dvd(string path, IFileSystem fileSystem)
         {
             _fileSystem = fileSystem;
             Titles = new List<Title>();
-            var allFiles = _fileSystem.GetFiles(path, true).ToList();
```

Comment:
This is going to reduce performance as we're now enumerating the list multiple times (lines: 24, 25 and 29)